lib/util/movingaveragedelta.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+/**
+ * Calculates the moving average delta for the given pair ofsamples. A sample (S)
+ * consists of a numerator (Sn) and a denominator (Sd).The moving average delta is
+ * calculated as follows:
+ *
+ * MovingAvgDelta = (Sn[1] - Sn[0]) / (Sd[1] - Sd[0])
+ */
+class MovingAverageDelta {
+  /**
+   * Constructor.
+   */
+  constructor() {
+    Object.defineProperties(this, {
+      _samples: {
+        value: [
+          { denominator: 0, numerator: 0 },
+          { denominator: 0, numerator: 0 }
+        ],
+      }
+    });
+  }
+
+  /**
+   * Get the moving average delta.
+   * @returns {number}
+   */
+  get() {
+    const { _samples: samples } = this;
+    const denominatorDelta = (samples[1].denominator - samples[0].denominator) || 1;
```

Comment:
I see that you are avoiding `divide by zero`  by replacing denominator delta with `1` . Would `Number.EPSILON` be more appropriate default?

Reply:
@makarandp0 The number calculated using `EPSILON` wouldn't make sense. Dividing by 1 is closer to what we want.

Reply:
you are right `EPSILON` is not good choice. But It does not look like 1 is right choice either, I think its better to return 0 from this function and  either a) not include 0 average values in the payload  (preferred) send to insights or b) report 0, but when querying in kibana we could exclude these 0 values.